lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1933,9 +1934,13 @@ sub handle_retry ($self) {
     return undef unless my $retry = $self->settings_hash->{RETRY};
     # strip any optional descriptions after a colon
     $retry =~ s/:.*//;
+    return 0 unless looks_like_number $retry;
     my $ancestors = $self->ancestors;
-    log_debug('handle_retry: Found retry job ' . $self->id . ", try '$ancestors' of up to '$retry' retries");
-    return $ancestors < $retry;
+    return 0 if $ancestors >= $retry;
+    my $system_user_id = $self->result_source->schema->resultset('Users')->system->id;
+    my $msg = "Restarting as RETRY set to $retry (and only restarted $ancestors times so far)";
```

Comment:
`Restarting as RETRY...` doesn't sound quite right to me. Maybe `Restarting because RETRY...`?